lib/commands/file.js

```diff
@@ -114,7 +113,54 @@ options(program, 'files-upload <file> <datasetID>')
         await upload(response.data.uploadUrl, file, fileSize);
         console.log(chalk.green(`Upload complete: ${fileName}, ID: ${response.data.id}`));
       }
+    } catch (error) {
+      if (_get(error, 'response.data.error', '').indexOf('already exists') > -1) {
+        console.log(chalk.yellow(`Ignoring already uploaded file: ${fileName}`));
+      } else {
+        throw error;
+      }
     }
-  });
 
-module.exports = program;
+    if (options.deleteAfterUpload) {
+      let verified = false;
+      for (let i = 1; i <= VERIFICATION_RETRIES; i++) {
+        await sleep(i * 500);
+
+        const searchOptions = {
+          datasetId: dataset,
+          name: fileName,
+          pageSize: 1
+        };
+        const searchResponse = await get(options, `${FILES_URL}?${querystring.stringify(searchOptions)}`);
```

Comment:
Originally, I was thinking we'd want to grab the files by ID for verification.  What's nice about this alternative approach is that it avoids conditional logic based on whether the file was uploaded (and we'd have a file ID) or the file was ignored because there is already one with the same name.